ecs.tf

```diff
@@ -115,6 +115,63 @@ variable "ecsInstancerolePolicy" {
 EOF
 }
 
+variable "use_ssm" {
+  default     = false
+  description = "Allow access via AWS Systems Mamager"
+}
+
+variable "ecsInstanceroleSsmPolicy" {
+  type = string
+
+  default = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ecs:CreateCluster",
+        "ecs:DeregisterContainerInstance",
+        "ecs:DiscoverPollEndpoint",
+        "ecs:Poll",
+        "ecs:RegisterContainerInstance",
+        "ecs:StartTelemetrySession",
+        "ecs:Submit*",
+        "ecr:GetAuthorizationToken",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:BatchGetImage",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+          "ec2messages:GetMessages",
+          "ssm:ListAssociations",
+          "ssm:ListInstanceAssociations",
+          "ssm:UpdateInstanceInformation",
+          "ssmmessages:CreateControlChannel",
+          "ssmmessages:CreateDataChannel",
+          "ssmmessages:OpenControlChannel",
+          "ssmmessages:OpenDataChannel"
+      ],
+      "Resource": "*"
```

Comment:
As above, any time an IAM permission is for resource `*`, it's worth looking to see if it can be narrowed down. As-is, this would allow that role to access any such applicable resources in that entire AWS account.

Reply:
That said, fine-tuning IAM permission can take a lot of time and experimentation, so it's a balancing act of how much time it's worth for how much benefit.